### DISCUSSION TITLE
How to skip join table in many-to-many relations in the schema?

### DISCUSSION THREAD

**Discussion by GP4cK** [TriPSs/nestjs-query#30] [Q&A]
Let's say you have a Project and a User entity with a many to many relation Project_User:

```ts
class Project {
  @OneToMany()
  links: Project_User[]
}

class User {
  @OneToMany()
  links: Project_User[]
}

class Project_User {
  @ManyToOne()
  project

  @ManyToOne()
  user

  // Other properties
}
```

The naive approach would be that you add a `@Relation` on top of each class and then you'll end up querying your project and users like this:

```graphql
query {
  project {
    links {
      user {
         id, name etc.
      }
    }
  }
}
```

Is there an easy way to do it like that instead?

```graphql
query {
  project {
    user {
       id, name etc.
    }
  }
}
```

I manage to make it work by making the ProjectService extend the RelationQueryService but it doesn't use the dataloader (so it's making on user database request for each project).
Have you ever faced this situation?

**Comment by TriPSs** [comment]
I think I partly know what you mean, but, you have a many to many but are using the ManyToOne?

**Reply by GP4cK** [reply]
Because my ManyToMany has additional properties. That's why I had to create a custom entity Project_User.
https://typeorm.io/many-to-many-relations#many-to-many-relations-with-custom-properties

**Reply by TriPSs** [reply]
Check, then I know what you mean, I have the same in my application and are handeling it the same way your are doing (first example)

**Comment by TriPSs** [comment]
Would like to point out that it would be awesome if we can figure out a way to make this work smoother, like that for the filtering it stays as is (so you can still use your custom fields) but have a option to directly (in your case) the user.
